tests/st/fuzz/check_pass_rate.py

```
@@ -0,0 +1,209 @@
+
+import argparse
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+_TRACEBACK_TAIL_LINES = 30
+
+
+def _parse_suites(xml_path: str) -> list[ET.Element]:
+    """Parse JUnit XML and return list of <testsuite> elements."""
+    root = ET.parse(xml_path).getroot()
```

Ruff (0.15.2)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)